src/mbgl/util/fnv_hash.hpp

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+namespace mbgl {
+namespace util {
+
+inline constexpr uint64_t hashFNV1a(const char * str, uint64_t value = 0xcbf29ce484222325) {
```

Comment:
This makes me wonder whether something like https://github.com/servo/string-cache would suit all our string usage needs.